### ISSUE TITLE
Installation w/ Containerized Home Assistant

### ISSUE THREAD

**Issue by sethgho** [wubbl0rz/FiatChamp#22]
Hi! I'm still pretty new to the Home Assistant ecosystem. The installation instructions say to install via Add-Ons. I'm running Home Assistant in a container, so the add-on system isn't available. Would it still be possible to install FiatChamp? I'm assuming I could spin it up in a separate adjacent container like I do zwave-js-ui and esphome, yea? 

I'd like to give this a go and test it against a Ram truck, if possible.

**Comment by wubbl0rz** [comment]
the container reads information provided by the supervisor at startup. mqtt server, token, options etc.  https://github.com/wubbl0rz/FiatChamp/blob/master/FiatChampAddon/run.sh 

this will fail if its not running as a addon. so i do not think the container will work standalone. to make it work standalone you would need to compile the .net program, build your own container and then provide all the environment variables normally set by the home assistant supervisor. so in theory it works but i would advise against it. maybe i should build a standalone container and offer it as an alternative install method.

**Comment by wubbl0rz** [comment]
running as standalone is supported in 3.0.4 which will be released soon :smile: 

example for docker compose:

``` yaml
version: "3.9"                                                                                                                                     
services:
  FiatChamp:
    image: ghcr.io/wubbl0rz/image-amd64-fiat-champ:3.0.4
    environment:
      - 'STANDALONE=True'
      - 'FiatChamp_FiatUser=user@example.com'
      - 'FiatChamp_FiatPw=123456'
      - 'FiatChamp_FiatPin=9999'
      - 'FiatChamp_SupervisorToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiI5NGFmMGJhZTFjYTQ0ODk2YWEzYjgzMGI5YmE4NGQxNiIsImlhdCI6MTY3MDA3Mjc
      - 'FiatChamp_StartDelaySeconds=1'
      - 'FiatChamp_Region=Europe'
      - 'FiatChamp_Brand=Fiat'
      - 'FiatChamp_CarUnknownLocation=Unknown'
      - 'FiatChamp_ConvertKmToMiles=False'
      - 'FiatChamp_MqttUser=mqtt'
      - 'FiatChamp_MqttPw=123456'
      - 'FiatChamp_MqttServer=192.168.2.132'
      - 'FiatChamp_MqttPort=1883'
```

**Comment by wubbl0rz** [comment]
3.0.4 can now be run as standalone container. see example above. i tried it locally and it works but maybe everything. if you have any problems just make a new issue :smile:

**Comment by SpykeKrepshaw** [comment]
> 
Did you ever get this working? I'm not sure how to set up a stand-alone version. Any help would be appreciated.

**Comment by wubbl0rz** [comment]
@SpykeKrepshaw  how does your docker compose file look ? its really not meant running standalone but should be possible.

**Comment by marcobto58** [comment]
HI
Congratulations for the excellent work.
I would like to migrate home assistant from the supervised version (works perfectly) to the one that uses containers.
Unfortunately I can't if I can't get Fiatchamp to work
I tried using the example configuration. I generated the token but I get this error:

ERROR: Call failed. Resource temporarily unavailable (supervisor:80): GET http://supervisor/core/api/states

see logs attached

[portainer.fiatchamp.log](https://github.com/wubbl0rz/FiatChamp/files/12579323/portainer.fiatchamp.log)


[_FiatChamp_logs.txt](https://github.com/wubbl0rz/FiatChamp/files/12575781/_FiatChamp_logs.txt)

**Comment by miklosbagi** [comment]
tl;dr: do we really have to run this in a docker container on the side?

Would something like this not work?

```
  homeassistant:
    image: "${HA_IMAGE:-homeassistant/home-assistant}:${HA_TAG:-latest}"
...
    volumes:
      # custom components (instead of HACS)
      - './custom_components/FiatChamp/FiatChampAddon
 ...
```

I'm running everything like this that'd end up in HACS and it works fine for these at least: `amaximus/fkf-garbage-collection`, `RobHofmann/HomeAssistant-GreeClimateComponent`, `nielsfaber/alarmo`, `rgc99/irrigation_unlimited`. HA finds these addons like they were added in a HAOS or Supervised install ¯\\\_(ツ)_\/¯
